src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,336 @@
+            sol_flow_send_error_packet(node, -r, "Could not disconnect to service: %s: %s",
+                     mdata->service_name, sol_util_strerrora(-r));
+    }
+
+    return 0;
+}
+
+static int
+connect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return connection_handle(node, data, true);
+}
+
+static int
+disconnect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return connection_handle(node, data, false);
+}
+
+static int
+get_service_state(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    struct sol_netctl_service *service;
+    int r;
+
+    service = find_service_by_name(mdata->service_name);
+    if (!service) {
+        sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+        return 0;
+    }
+    r = sol_flow_send_string_packet(node,
+             SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__OUT,
+             sol_netctl_service_state_to_str(mdata->state));
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (mdata->state == SOL_NETCTL_SERVICE_STATE_ONLINE)
+            r = sol_flow_send_bool_packet(node,
+                         SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+                         true);
+    else
+            r = sol_flow_send_bool_packet(node,
+                         SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+                         false);
+
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static int
+set_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    const char *in_value;
+    int r;
+
+    r = sol_flow_packet_get_string(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mdata->service_name = strdup(in_value);
+    SOL_NULL_CHECK(mdata->service_name, -ENOMEM);
+
+    return 0;
+}
+
+static int
+open_network_service(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct network_service_data *mdata = data;
+    const struct sol_flow_node_type_netctl_service_options *opts;
+    int r;
+
+    SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options,
+        SOL_FLOW_NODE_TYPE_NETCTL_SERVICE_OPTIONS_API_VERSION,
+        -EINVAL);
+    opts = (const struct sol_flow_node_type_netctl_service_options *)
+        options;
+
+    if (opts->name) {
+        mdata->service_name = strdup(opts->name);
+        SOL_NULL_CHECK(mdata->service_name, -ENOMEM);
+    }
+
+    mdata->state = SOL_NETCTL_SERVICE_STATE_UNKNOWN;
+
+    r = sol_netctl_add_service_monitor(service_status_cb, node);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_netctl_add_error_monitor(error_cb, node);
+    SOL_INT_CHECK(r, < 0, r);
+
```

Comment:
I think you should force a `TICK` here by calling `get_service_state()` if there is a desired service name.

For connman implementation it will happen as magic, since the DBus call will deliver results later, that will trigger the "service_status_cb". But for other implementations or even if the FBP was loaded later in the process lifetime, the services list will be known and thus nothing will be dispatched unless the service changes.

Forcing an initial "TICK" (`get_service_state()`) will produce the same result every time. It's consistent.